tests/common/test_cgroupconfigurator.py

```diff
@@ -46,63 +48,63 @@ def tearDownClass(cls):
         CGroupConfigurator._instance = None  # pylint: disable=protected-access
         AgentTestCase.tearDownClass()
 
-    @staticmethod
-    def _get_new_cgroup_configurator_instance(initialize=True, mock_commands=None, mock_files=None):
+    @contextlib.contextmanager
+    def _get_cgroup_configurator(self, initialize=True, mock_commands=None, mock_files=None):
```

Comment:
Previously the scope of the mocks was limited to the initialize() method, which makes some tests hard to write.

Returning a context manager allows the caller to take control of the scope.